common/src/main/java/org/keycloak/common/enums/SslRequired.java

```diff
@@ -35,23 +35,29 @@ public boolean isRequired(ClientConnection connection) {
         return isRequired(connection.getRemoteAddr());
     }
 
-    public boolean isRequired(String address) {
+    public boolean isRequired(String host) {
```

Comment:
This should be an address no? and not a host?

Reply:
Unfortuantely it is not, we currently call this method from other places to validate URLs - see the comment above https://github.com/keycloak/keycloak/pull/33530/files/bf191bcdbd410cb809a3d94458ff6aa422e6de0a#diff-0515f8371b64463d3e1a2f81c0bad1a262ab7003430322956fc3d9ffd89a1748R45

However that usage seems dubious as well.

Reply:
Yer, looking at UriUtils that is weird. SslRequired was built to check if SSL was required when accessing a realm, not really intended to check if SSL was required to external connections.

What about

Reply:
Yer, it's getting kinda messy this.

Looks like we essentially have two use-cases here:

* isRequired(ClientConnection connection) - check if TLS is required for a client connection. Here we could do a InetAddress.getByAddress. However, how do you convert a string to byte[] in an elegant way?
* isRequired(String host) - check if TLS is required for connecting to external things (i.e. IdP brokering). Here it's okay to do getByName as it's URLs configured by an admin, and there will be a DNS lookup regardless as the request is sent

Reply:
> Maybe, I'm overthinking this part though, and it's not really worth the effort.

With the guarentee that getRemoteAddr returns an address, not a hostname, it didn't seem to need additional effort.